Inkscape fails to build with

```

inkscape> [381/1121] Building CXX object src/CMakeFiles/inkscape_base.dir/extension/internal/pdfinput/svg-builder.cpp.o
inkscape> /build/inkscape-1.4.2_2025-05-08_ebf0e940d0/src/extension/internal/pdfinput/svg-builder.cpp: In member function 'Inkscape::XML::Node* Inkscape::Extension::Internal::SvgBuilder::_createImage(Stream*, i:
inkscape> /build/inkscape-1.4.2_2025-05-08_ebf0e940d0/src/extension/internal/pdfinput/svg-builder.cpp:2035:28: warning: ignoring return value of 'bool ImageStream::reset()', declared with attribute 'nodiscard' ]
inkscape>  2035 |         image_stream->reset();
inkscape>       |         ~~~~~~~~~~~~~~~~~~~^~
inkscape> In file included from /nix/store/9izqbypbvx6vvdyay9269vz8jhrfvw4b-poppler-glib-25.06.0-dev/include/poppler/Object.h:605,
inkscape>                  from /nix/store/9izqbypbvx6vvdyay9269vz8jhrfvw4b-poppler-glib-25.06.0-dev/include/poppler/GfxState.h:44,
inkscape>                  from /nix/store/9izqbypbvx6vvdyay9269vz8jhrfvw4b-poppler-glib-25.06.0-dev/include/poppler/Gfx.h:40,
inkscape>                  from /build/inkscape-1.4.2_2025-05-08_ebf0e940d0/src/extension/internal/pdfinput/svg-builder.h:30,
inkscape>                  from /build/inkscape-1.4.2_2025-05-08_ebf0e940d0/src/extension/internal/pdfinput/svg-builder.cpp:17:
inkscape> /nix/store/9izqbypbvx6vvdyay9269vz8jhrfvw4b-poppler-glib-25.06.0-dev/include/poppler/Stream.h:474:24: note: declared here
inkscape>   474 |     [[nodiscard]] bool reset();
inkscape>       |                        ^~~~~
inkscape> /build/inkscape-1.4.2_2025-05-08_ebf0e940d0/src/extension/internal/pdfinput/svg-builder.cpp:2061:28: warning: ignoring return value of 'bool ImageStream::reset()', declared with attribute 'nodiscard' ]
inkscape>  2061 |         image_stream->reset();
inkscape>       |         ~~~~~~~~~~~~~~~~~~~^~
inkscape> /nix/store/9izqbypbvx6vvdyay9269vz8jhrfvw4b-poppler-glib-25.06.0-dev/include/poppler/Stream.h:474:24: note: declared here
inkscape>   474 |     [[nodiscard]] bool reset();
inkscape>       |                        ^~~~~
inkscape> ninja: build stopped: subcommand failed.
```

--- a/pkgs/applications/graphics/inkscape/default.nix
+++ b/pkgs/applications/graphics/inkscape/default.nix
@@ -97,6 +97,16 @@ stdenv.mkDerivation (finalAttrs: {
       # Fix path to ps2pdf binary
       inherit ghostscript;
     })
+    (fetchpatch {
+      name = "fix-build-poppler-25.06.0.patch";
+      url = "https://gitlab.com/inkscape/inkscape/-/commit/97bd8f29a61e691ceea98ca2444b974cf4256ae0.patch";
+      hash = "sha256-bYRd/KUh/7qFb7x0EuUgQYA9P8abcTf5XS67gzaAiXA=";
+    })
+    (fetchpatch {
+      name = "fix-build-poppler-25.07.0.patch";
+      url = "https://gitlab.com/inkscape/inkscape/-/commit/ce52c5f96106ae5747171663a46831f21aa52d95.patch";
+      hash = "sha256-3Yj+neSRSSQPeeZkHJ0P6v3Sis/lg9xiygktI6Z+zDY=";
+    })
   ];
 
   postPatch = ''

--- a/pkgs/applications/office/scribus/default.nix
+++ b/pkgs/applications/office/scribus/default.nix
@@ -105,6 +105,16 @@ stdenv.mkDerivation (finalAttrs: {
       url = "https://aur.archlinux.org/cgit/aur.git/plain/fix_build_with_poppler_25.02.0.patch?h=scribus-unstable";
       hash = "sha256-t9xJA6KGMGAdUFyjI8OlTNilewyMr1FFM7vjHOM15Xg=";
     })
+    (fetchpatch {
+      name = "fix-build-poppler-25.06.0.patch";
+      url = "https://github.com/scribusproject/scribus/commit/8dcf8d777bd85a0741c455961f2de382e3ed47ec.patch";
+      hash = "sha256-JBHCgvEJnYrUdtLnFSXTfr1FFin4uUNUnddYwfRbn7k=";
+    })
+    (fetchpatch {
+      name = "fix-build-poppler-25.07.0.patch";
+      url = "https://github.com/scribusproject/scribus/commit/ff6c6abfa8683028e548a269dee6a859b6f63335.patch";
+      hash = "sha256-N4jve5feehsX5H0RXdxR4ableKL+c/rTyqCwkEf37Dk=";
+    })
   ];
 
   meta = {

--- a/pkgs/development/libraries/poppler/default.nix
+++ b/pkgs/development/libraries/poppler/default.nix
@@ -55,13 +55,13 @@ let
     domain = "gitlab.freedesktop.org";
     owner = "poppler";
     repo = "test";
-    rev = "91ee031c882634c36f2f0f2f14eb6646dd542fb9";
-    hash = "sha256-bImTdlhMAA79kwbKPrHN3a9vVrtsgBh3rFjH3B7tEbQ=";
+    rev = "c79c6839e859dbee6b73ac260788fa2de8618ba4";
+    hash = "sha256-j66AsBUnFpO5athVgQmf4vcyXxYcJ/plJtHg+3vXG4Y=";
   };
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "poppler-${suffix}";
-  version = "25.05.0"; # beware: updates often break cups-filters build, check scribus too!
+  version = "25.07.0"; # beware: updates often break cups-filters build, check scribus too!
 
   outputs = [
     "out"
@@ -70,7 +70,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   src = fetchurl {
     url = "https://poppler.freedesktop.org/poppler-${finalAttrs.version}.tar.xz";
-    hash = "sha256-mxYnxbdoFqxeQFKgP1tgW6QLRc8GsCyt0EeWILSZqzg=";
+    hash = "sha256-xQSpBm29/r43etU87GQf2XHulsTh6Mp05snAPUbYF64=";
   };
 
   nativeBuildInputs =

